### PR TITLE
centralise client setup and access of api key

### PR DESCRIPTION
**Pull request by alex-straw**
## Overview
I've refactored the OpenAI client initialisation into a single module to clean up the codebase and make it more DRY.

Following this (if people would like), I want to add a mock OpenAI client (which can be used if a specific env var is set). This addition will enable contributors to test the project more easily and without a valid API key. The centralised client setup change should make adding this mock client more straightforward.

## PR Feedback
I've introduced pydantic-settings for environment variable management (suggsted by @gruckion - see commit: [link](https://github.com/daveshap/OpenAI_Agent_Swarm/pull/76/commits/29daff94ee766201731e0e9df30ab939b7ecea00)). I'm open to any feedback or concerns regarding the introduction of this new library (requirements.txt soon?). 

Pydantic-settings streamlines our setup and eases the addition of APIs like Pinecone in the future, a point @daveshap touched on in his recent video about enhancing agent communication.

### CODE REVIEW

**Comment by RomanGoEmpire** [comment]
looks good 👍 Can be merged

File: shared/openai_config.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import os
+import dotenv
+from openai import OpenAI
+
+dotenv.load_dotenv()
+
+def get_openai_client():
```
Comment (gruckion) [inline shared/openai_config.py:7]: Best if you create a central place to load env vars from. Can use pydantic for class

Comment (alex-straw) [reply shared/openai_config.py:4]: @gruckion I've created a shared/settings.py file which uses Pydantic for this, let me know what you think.

**Comment by daveshap** [comment]
It looks like this conflicts with another PR. Please take a look and rectify. Let's get these fixes merged!

**Comment by alex-straw** [comment]
Merge conflicts resolved @daveshap, and I've also added in ```client = get_openai_client()``` in a few more places where I spotted we were setting up clients.

**Pull request merged**